var/spack/repos/builtin/packages/neurodamus-core/package.py

```diff
@@ -42,17 +46,44 @@ def install(self, spec, prefix):
         shutil.copytree('mod', prefix.mod)
         if spec.satisfies('+python'):
             copy_tree('resources/neurodamus-py', prefix.python)
-        if spec.satisfies('+common'):
-            copy_all('resources/common/hoc', prefix.hoc)
-            copy_all('resources/common/mod', prefix.mod)
 
         filter_file(r'UNKNOWN_CORE_VERSION', r'%s' % spec.version, prefix.hoc.join('defvar.hoc'))
-
         try:
             commit_hash = self.fetcher[0].get_commit()
             filter_file(r'UNKNOWN_CORE_HASH', r'\'%s\'' % commit_hash, prefix.hoc.join('defvar.hoc'))
         except Exception as e:
-            tty.warn(e)
+            tty.warn(str(e))
+
+        # +Common will bring common mods and build a bare nrnmechlib
+        if spec.satisfies('+common'):
+            copy_all('resources/common/hoc', prefix.hoc)
+            copy_all('resources/common/mod', prefix.mod)
+
+            with working_dir(prefix):
+                which('nrnivmodl')('-incflags', '-DDISABLE_REPORTINGLIB', 'mod')
+
+                bindir = os.path.basename(self.neuron_archdir)
+                special = join_path(bindir, 'special')
+                open('.bindir', 'w').write(bindir)
+
+                if not os.path.isfile(special):
+                    raise Exception("Failed to build neurodamus core mods")
+
+                # Cleanup
+                for f in find(bindir, '*.lo'): os.remove(f)
+                for f in find(bindir, '*.mod'): os.remove(f)
+                for f in find(bindir + '/.libs', '*.o'): os.remove(f)
+                os.mkdir(bindir + "/mod2c")
+                for f in find(bindir, "*.c*"): shutil.move(f, bindir + "/mod2c/")
```

Comment:
why we need mod2c here?

Reply:
debugging? we have been keeping them in all neurodamus... I moved them to x86_64/mod2c to cleanup a bit, but I could delete...

Reply:
I meant, it's not mod2c. (mod2c directory is typically for CoreNEURON).

Reply:
I'll rename like in other neurodamus, to `modc`